llama-index-integrations/indices/llama-index-indices-managed-postgresml/llama_index/indices/managed/postgresml/base.py

```diff
@@ -0,0 +1,198 @@
+    """
+
+    def __init__(
+        self,
+        collection_name: str,
+        pipeline_name: Optional[str] = None,
+        pipeline_schema: Optional[Dict[str, Any]] = None,
+        pgml_database_url: Optional[str] = None,
+        show_progress: bool = True,
+        upsert_parallel_batches: int = 1,
+        nodes: Optional[Sequence[BaseNode]] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the PostgresML SDK."""
+        self.show_progress = show_progress
+        self.upsert_parallel_batches = upsert_parallel_batches
+
+        index_struct = PostgresMLIndexStruct(
+            index_id=collection_name,
+            summary="PostgresML Index",
+        )
+
+        super().__init__(
+            show_progress=show_progress,
+            index_struct=index_struct,
+            **kwargs,
+        )
+
+        # Create our Collection and Pipeline
+        self.collection = Collection(collection_name, pgml_database_url)
+        if pipeline_name is None:
+            pipeline_name = "v1"
+        if pipeline_schema is None:
+            pipeline_schema = {
+                "content": {
+                    "splitter": {
+                        "model": "recursive_character",
+                        "parameters": {"chunk_size": 1500},
+                    },
+                    "semantic_search": {
+                        "model": "intfloat/e5-small-v2",
+                        "parameters": {"prompt": "passage: "},
+                    },
+                }
+            }
+        self.pipeline = Pipeline(pipeline_name, pipeline_schema)
+
+        # We must wrap self.collection.add_pipeline() with this async function
+        # This is a limitation of the pyo3 async implementation
+        async def add_pipeline():
+            await self.collection.add_pipeline(self.pipeline)
+
+        run_async_tasks([add_pipeline()])
+
+        if nodes:
+            self._insert(nodes)
+
+    def _insert(
+        self,
+        nodes: Sequence[BaseNode],
+        **insert_kwargs: Any,
+    ) -> None:
+        """Insert a set of documents (each a node)."""
+        documents = [
+            {
+                "id": node.node_id,
+                "content": node.get_content(),
+                "metadata": node.metadata,
+            }
+            for node in nodes
+        ]
+
+        args = {"parallel_batches": self.upsert_parallel_batches, **insert_kwargs}
+
+        # We must wrap self.collection.upsert_documents() with this async function
+        # This is a limitation of the pyo3 async implementation
+        async def upsert_documents():
+            await self.collection.upsert_documents(documents, args)
+
+        run_async_tasks([upsert_documents()])
+
+    def add_documents(
+        self,
+        docs: Sequence[Document],
+        **insert_kwargs: Any,
+    ) -> None:
+        nodes = [
+            TextNode(id_=doc.id_, text=doc.get_content(), metadata=doc.metadata)
```

Comment:
A document and a text node are nearly the same object (its mostly just what they represent)

You can more easily convert between the two using `node = TextNode(**doc.dict())`